### DISCUSSION TITLE
Supabase returning an empty list for the foreign key relation

### DISCUSSION THREAD

**Discussion by FezVrasta** [supabase/supabase#14394] [Questions]
I have two tables, users (id, name), and services (id, user_id, name).

The `services.user_id` column is set with a foreign key relation to `users.id`.

In Supabase studio I see the relation working, if I click the right arrow next to a `user_id` it brings me to the correct `user` entry.

I need to query a user and all its related services from Supabase JS now, and I'm using:

```js
  const { data, error } = await supabase
    .from('users')
    .select('*, services(*)')
    .eq('id', id)
    .single();
```

The problem is that this returns me the correct user object, but with an empty array as `services`:

```
{ id: 'xxxx', name: 'Foobar', services: [] }
```

These are my table definitions:
```sql
create table
  public.services (
    id uuid not null,
    created_at timestamp with time zone null default now(),
    constraint users_id_pkey primary key (id)
  ) tablespace pg_default;

create table
  public.services (
    id uuid not null,
    user_id uuid not null,
    created_at timestamp with time zone not null default now(),
    name text not null,
    constraint services_pkey primary key (id),
    constraint services_user_id_fkey foreign key (user_id) references users (id) on delete cascade
  ) tablespace pg_default;
```

What am I doing wrong? I have data in the database that should be returned.

**Comment by agboolaidris** [comment]
I am facing the same issue

**Comment by PcorreAc** [comment]
were you able to solve the problem?
